File: csharp/sdk/src/ModelContextProtocol.Interceptors/Gateway/McpInterceptorServerConnectionOptions.cs
using Microsoft.Extensions.Logging;
using ModelContextProtocol.Client;

namespace ModelContextProtocol.Interceptors.Gateway;

/// <summary>
/// Configuration for connecting the gateway to an external SEP-exposing interceptor server
/// using the standard MCP client transport patterns.
/// </summary>
public sealed class McpInterceptorServerConnectionOptions
{
    /// <summary>
    /// Gets or sets an optional stable identifier for reusing a connected interceptor client across requests.
    /// When omitted, the gateway creates and disposes a client for each resolution.
    /// </summary>
    public string? ConnectionId { get; set; }

    /// <summary>
    /// Gets or sets the client transport used to connect to the external interceptor server.
    /// </summary>
    public required IClientTransport Transport { get; set; }

    /// <summary>
    /// Gets or sets optional MCP client configuration used when creating the interceptor client.
    /// </summary>
    public McpClientOptions? ClientOptions { get; set; }

    /// <summary>
    /// Gets or sets an optional logger factory used when creating the interceptor client.
    /// </summary>
    public ILoggerFactory? LoggerFactory { get; set; }
}

File: typescript/sdk/src/gateway/mcp-interceptor-server-connection-options.ts
// Copyright 2025 The MCP Interceptors Authors. All rights reserved.
import type { Client, Implementation } from "@modelcontextprotocol/client";

export type InterceptorClientTransport = Parameters<Client['connect']>[0];

/** Options for connecting the gateway to an external interceptor host. */
export interface McpInterceptorServerConnectionOptions {
  /**
   * Stable id for reusing a connected client across requests.
   * When omitted, a client is created and closed after each resolution.
   */
  connectionId?: string;
  transport: InterceptorClientTransport;
  clientInfo?: Implementation;
  clientOptions?: ConstructorParameters<typeof Client>[1];
}
